### PR TITLE
1064 add local document downloader

### PR DESCRIPTION
**Pull request by leite08**
Ref: metriport/metriport-internal#1064

### Dependencies

none

### Description

- add local document downloader
- lambda to use local document downloader
- make factory to create document downloader
   - returns local for local environment
   - returns lambda for cloud env
- api to get document downloader from factory
- move OID code to core

### Release Plan

- staging
   - [x] publish core
   - [x] update lambda
   - [ ] merge this
- production
   - [ ] publish core
   - [ ] update lambda
   - [ ] merge this

### CODE REVIEW

**Review by RamilGaripov — APPROVED** [review]
LGTM 🚀

File: packages/api/src/external/commonwell/document/document-downloader-factory.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+  return new DocumentDownloaderLambda({
+    region,
+    bucketName,
+    lambdaName: Config.getDocumentDownloaderLambdaName(),
+    orgName,
+    orgOid,
+    npi,
+  });
+}
```
Comment (Goncharo) [inline packages/api/src/external/commonwell/document/document-downloader-factory.ts:40]: 😍

File: packages/core/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@metriport/core",
-  "version": "1.3.0-alpha.31",
+  "version": "1.3.0-alpha.1",
```
Comment (Goncharo) [inline packages/core/package.json:3]: was this intentional? looks like it's going back a version

File: packages/core/src/external/commonwell/document/document-downloader-local.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+      bucket: downloadResult.bucket,
+      key: downloadResult.key,
+      location: downloadResult.location,
+      size: downloadResult.size,
+      contentType: downloadResult.contentType,
+    };
+
+    if (
+      downloadedDocument &&
+      (document.mimeType === "application/xml" || document.mimeType === "text/xml")
```
Comment (Goncharo) [inline packages/core/src/external/commonwell/document/document-downloader-local.ts:68]: nit: could put the check into a function in our `core` package like:

```
mimeTypeIsXML(mimeType) {
   return mimeType === "application/xml" || mimeType === "text/xml");
}
```

this way we will never forget to check for both mimetypes, I remember we had an issue recently where we forgot to check for `text/xml` and it broke some logic


also, we could but `"text/xml"` and etc into the core package, and re-use those string consts (I see 2 other places where it's used in this PR)

**Review by Goncharo — APPROVED** [review]
❤️ the approach - a couple nits/questions to consider

File: packages/core/package.json (hunk 2 above)
Comment (leite08) [reply packages/core/package.json:3]: Ditto diff branches work8ng on the same next version

**Pull request merged**